src/mqt/predictor/ml/predictor.py

```diff
@@ -287,14 +292,21 @@ def train_random_forest_classifier(
             },
         ]
 
-        clf = RandomForestClassifier(random_state=0)
+        if device is None:  # Default classification model
```

Comment:
Shouldn't we rather first check for the set figure of merit and if it is the helliger distance, check that the device is set? This currently seems a little unintuitive since it is based on the assumption, that never a device is provided for another figure of merit - which could happen if user misinterpret the parameters.